packages/bindgen/src/templates/jsi.ts

```diff
@@ -946,6 +951,52 @@ class JsiCppDecls extends CppDecls {
           throw jsi::JSError(_env, "Can't convert " + val.toString(_env).utf8(_env) + " to Mixed");
         `,
       }),
+      new CppFunc("bigIntToI64", "int64_t", [new CppVar("jsi::Runtime&", env), new CppVar("auto&&", "val")], {
+        body: `
+          if (val.isBigInt())
+              return FWD(val).asBigInt(_env).asInt64(_env);
+
+          auto obj = FWD(val).asObject(_env);
+          auto high = uint32_t(obj.getProperty(_env, ${this.addon.getPropId("high")}).asNumber());
+          auto low = uint32_t(obj.getProperty(_env, ${this.addon.getPropId("low")}).asNumber());
+          return int64_t((uint64_t(high) << 32) | low);
+        `,
+      }),
+      new CppFunc("bigIntToU64", "uint64_t", [new CppVar("jsi::Runtime&", env), new CppVar("auto&&", "val")], {
+        body: `
+          if (val.isBigInt())
+              return FWD(val).asBigInt(_env).asUint64(_env);
+
+          auto obj = FWD(val).asObject(_env);
+          auto high = uint32_t(obj.getProperty(_env, ${this.addon.getPropId("high")}).asNumber());
+          auto low = uint32_t(obj.getProperty(_env, ${this.addon.getPropId("low")}).asNumber());
+          return (uint64_t(high) << 32) | low;
+        `,
+      }),
+      new CppFunc("bigIntFromI64", "jsi::Value", [new CppVar("jsi::Runtime&", env), new CppVar("int64_t", "val")], {
+        body: `
+          if (${this.addon.get()}->m_on_hermes)
+              return jsi::BigInt::fromInt64(_env, val);
+
+          return ${this.addon.accessCtor("Long")}.callAsConstructor(_env, {
+              jsi::Value(int(val)), // low
+              jsi::Value(int(val >> 32)), // high
+              jsi::Value(false), // unsigned
+          });
+        `,
+      }),
+      new CppFunc("bigIntFromU64", "jsi::Value", [new CppVar("jsi::Runtime&", env), new CppVar("uint64_t", "val")], {
+        body: `
+          if (${this.addon.get()}->m_on_hermes)
+              return jsi::BigInt::fromUint64(_env, val);
+
+          return ${this.addon.accessCtor("Long")}.callAsConstructor(_env, {
+              jsi::Value(int(val)), // low
+              jsi::Value(int(val >> 32)), // high
+              jsi::Value(true), // unsigned
+          });
+        `,
```

Comment:
If at all practical, I would love to see snippets like these maintained outside of a .ts file to get better editor support. I do see however it relies on values being injected into it, so that might not be practical here.

Reply:
I initially started writing the first two in the jsi helpers header, but then moved them here when I realized that I wanted to use the `getPropId` mechanism to cache the `high` and `low` property constructions. I don't have a great answer here unfortunately. This seems like the least bad option, and isn't that bad in practice.